app/src/components/avatar_event_bus.rs

```diff
@@ -0,0 +1,55 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use yew_agent::{Agent, AgentLink, Context, HandlerId};
+
+use super::avatar::AvatarData;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Request {
```

Comment:
No need to make them enums, no?

Reply:
Actually the request can be an enum: one to request async fetching, and one to just notify of an update.